### ISSUE TITLE
Support Mono Track

### ISSUE THREAD

**Issue by rygantic217** [ZL-Audio/ZLEqualizer#51]
Hi i just recently found this plug-in from a youtube video and was interested in trying it out. Tried to download it a few weeks ago and upon opening Logic I got a message along the lines of "ZLEqualizer cannot be opened because the developer cannot be verified". I figured I'd try again today and after downloading and opening up Logic I got a message that said "ZLEqualizer can't be opened because Apple can't check it for malicious software". Following that message it says "This software needs to be updated. Contact the developer for more information". The message doesn't pop up every time I open Logic, but in the Plug-in Manager it says cannot be opened under ZLEqualizer. Was wondering how I can fix this. I am running Logic 10.8.1 and Sonoma 14.4.1.

**Comment by mikelupe** [comment]
Hi - The developer has stated this in https://github.com/ZL-Audio/ZLEqualizer/wiki/Manual

It is expected.

**Comment by rygantic217** [comment]
Thank you but what is expected? The pop-up message? I'm not sure where that is being referred to in the manual and I'm not sure how to get the plug-in to open because that message doesn't even pop up anymore. Also I don't know why it says the software needs to be updated and to contact the developer. Thats why i was making this post

**Comment by mikelupe** [comment]
I meant the plugin not having been notarized, sorry for not being clear enough. Maybe you clicked on some "do not trust" button, so it doesn't appear anymore. Let's see what the dev says, I don't have a Mac.

**Comment by zsliu98** [comment]
Please read through the following documentation from Apple:

https://support.apple.com/en-us/102445

and the following statement from the manual:

> You should decide whether to trust the installers from the provided link. Alternatively, you can build the plugin from the project source code.

**Comment by rygantic217** [comment]
> I meant the plugin not having been notarized, sorry for not being clear enough. Maybe you clicked on some "do not trust" button, so it doesn't appear anymore. Let's see what the dev says, I don't have a Mac.

Got it, no worries. Thanks for the clarification. Yes i dont remember what the option was but one was cancel and the other one just said like show in finder

**Comment by rygantic217** [comment]
> Please read through the following documentation from Apple:
> 
> https://support.apple.com/en-us/102445
> 
> and the following statement from the manual:
> 
> > You should decide whether to trust the installers from the provided link. Alternatively, you can build the plugin from the project source code.
> 
> I will also attach the above link to the manual. Thanks for reaching out.

@zsliu98 Thank you. I went to the Privacy and Security settings but i don't see the message for the open anyway option. is there a way i can get it to show up again? I tried deleting and downloading it again and no message popped up

**Comment by zsliu98** [comment]
> > Please read through the following documentation from Apple:
> > https://support.apple.com/en-us/102445
> > and the following statement from the manual:
> > > You should decide whether to trust the installers from the provided link. Alternatively, you can build the plugin from the project source code.
> > 
> > 
> > I will also attach the above link to the manual. Thanks for reaching out.
> 
> @zsliu98 Thank you. I went to the Privacy and Security settings but i don't see the message for the open anyway option. is there a way i can get it to show up again? I tried deleting and downloading it again and no message popped up

@rygantic217 You may use [PluginDoctor](https://ddmf.eu/plugindoctor/) (the demo version is enough for this) to open the plugin manually and hopefully the message will pop up again.

**Comment by rygantic217** [comment]
> > > Please read through the following documentation from Apple:
> > > https://support.apple.com/en-us/102445
> > > and the following statement from the manual:
> > > > You should decide whether to trust the installers from the provided link. Alternatively, you can build the plugin from the project source code.
> > > 
> > > 
> > > I will also attach the above link to the manual. Thanks for reaching out.
> > 
> > 
> > @zsliu98 Thank you. I went to the Privacy and Security settings but i don't see the message for the open anyway option. is there a way i can get it to show up again? I tried deleting and downloading it again and no message popped up
> 
> @rygantic217 You may use [PluginDoctor](https://ddmf.eu/plugindoctor/) (the demo version is enough for this) to open the plugin manually and hopefully the message will pop up again.

@zsliu98 Thank you! This did work, the message has popped up again and i was able to follow the instructions through the previous post with Apple. However, after i try to select the file to load the plug-in in Plugin Doctor I now get the message "An Audio Unit plug-in reported a problem which might cause the system to become unstable. Please quit and restart Logic Pro". I tried quitting and restarting Logic Pro, and also went to the plug-in manager and selected the plug-in. Any idea on what to do? (sorry for all the responses if this is a common issue, I wasn't expecting to run into this many obstacles and was unsure how to fix)

**Comment by zsliu98** [comment]
If you can load the plugin in Plugin Doctor and see the response curve when a band is added, the plugin should be functional. Unfortunately, I am not experienced with Logic Pro. I have just Google the message. It seems that it is specific to M* chip and different people have different answers to that.

Some links that might help:

https://support.apple.com/en-us/109315

https://discussions.apple.com/thread/254896459?sortBy=best&page=1

https://forum.juce.com/t/logic-10-7-8-and-native-m1-crash-on-some-systems/56495/3

https://www.reddit.com/r/Logic_Studio/comments/1bya4h4/i_keep_getting_this_error_message_anyone_know_how/

**Comment by rygantic217** [comment]
@zsliu98 thank you so much for the quick, detailed responses and helpful links. When running the plug-in again through Plugin Doctor, the interface of the ZLEqualizer loads and pops up for about half a second and then instantly causes Logic to crash :(. I tried to run Rosetta as suggested in the Reddit post but that didn't work either. Oh well. I look forward to trying any new plug-ins you develop or maybe an update can fix it in the future! I understand if you cannot due to your unfamiliarity of Logic Pro, I'm sure it's difficult to keep up with all this stuff and software updates

**Comment by zsliu98** [comment]
Thanks for your patience. In such a case, I would suspect that there are some bugs in my code (at least in version 0.1.16) or in JUCE. I will look into it when I have time (probably after finishing UI setting panel & static gain compensation).

**Comment by rygantic217** [comment]
@zsliu98 Hey I actually just tried again to see if it worked (because i found it so odd that it shows up as succesfully validated in the plugin manager but wasn't showing up under Audio Units) and I was able to get it to show up under Audio Units and get it to work. However, it only shows up in Audio Units if it is on a stereo track and is only able to be used on stereo tracks. Is there a way to get it to work on mono tracks as well or was this occurence by design?

**Comment by zsliu98** [comment]
> @zsliu98 Hey I actually just tried again to see if it worked (because i found it so odd that it shows up as succesfully validated in the plugin manager but wasn't showing up under Audio Units) and I was able to get it to show up under Audio Units and get it to work. However, it only shows up in Audio Units if it is on a stereo track and is only able to be used on stereo tracks. Is there a way to get it to work on mono tracks as well or was this occurence by design?

@rygantic217  Yes, it is by design and the plugin will tell the host that mono is not supported. The plugin requires stereo main input & side input.
I will add an inefficient workaround so that it will accept mono input (probably after UI setting & static gain compensation). I will let this issue open and inform you once I complete it.

**Comment by rygantic217** [comment]
@zsliu98 Awesome! Thanks so much for the update and I'm glad we found the root of the problem and got it to work! Look forward to it

**Comment by zsliu98** [comment]
Hi! The pre-release 0.2.0 should support the mono track. Would you like to verify it if you have time?

**Comment by rygantic217** [comment]
@zsliu98 Hi, thanks for getting back to me. I will try tmrw and let you know!

**Comment by rygantic217** [comment]
@zsliu98 just tried it, the mono track now works! However, there is a problem with the frequency band showing. The colored dot loads but when you move it the bands aren't there, so I can't see the visual change of the band or the Q. Hopefully this is an easy fix

**Comment by zsliu98** [comment]
> @zsliu98 just tried it, the mono track now works! However, there is a problem with the frequency band showing. The colored dot loads but when you move it the bands aren't there, so I can't see the visual change of the band or the Q. Hopefully this is an easy fix

@rygantic217 That sounds problematic. If you can't see the curves, the filters are not working. However, I am not able to reproduce it locally. Could you please tell me whether the plugin is enabled when you drag the buttons?

**Comment by rygantic217** [comment]
@zsliu98 so I just tried again and on a stereo track. The curve actually shows up on the mono track when the audio is playing. However, if the audio is not playing, you get the bug where you move the dot and only the dot moves and you don’t see the curve/frequency band. On the stereo track it works as intended and the curves show up. You are able to see and move the curve and see the Q adjustment while audio is playing AND when audio is not playing. 

So to recap: it seems that the mono track has an issue with not showing the band when audio is NOT playing in Logic. When audio IS playing, the band does show up. Hope this helps boil down the problem!

**Comment by zsliu98** [comment]
> @zsliu98 so I just tried again and on a stereo track. The curve actually shows up on the mono track when the audio is playing. However, if the audio is not playing, you get the bug where you move the dot and only the dot moves and you don’t see the curve/frequency band. On the stereo track it works as intended and the curves show up. You are able to see and move the curve and see the Q adjustment while audio is playing AND when audio is not playing.
> 
> So to recap: it seems that the mono track has an issue with not showing the band when audio is NOT playing in Logic. When audio IS playing, the band does show up. Hope this helps boil down the problem!

@rygantic217 Thanks for testing. It looks like Logic Pro is very *smart* and it completely stops the plugin processor when the playback is paused (only on mono track). I will add a workaround (probably in the formal release of 0.2.0) but don't put much hope on that :sweat:

**Comment by rygantic217** [comment]
@zsliu98 no problem! Thanks for the response

**Comment by zsliu98** [comment]
> @zsliu98 no problem! Thanks for the response

Hi. Release 0.2.0 may resolve this issue :worried: BTW, I have also changed the installer from `.dmg` to `.pkg`. Now the warning message shows up when you open the installer. Hopefully, it will make the installation process more friendly.

**Comment by rygantic217** [comment]
@zsliu98 just downloaded. Yes, this process made the installation much more user friendly! Will definitely help others be able to successfully download the plug-in. I tried out the mono track issue and it is still the same unfortunately :/. I will continue using it with the playback playing since it shows the frequency band then. I tried out a stereo track to compare and also make sure everything was working for stereo track and stereo track worked fine.

**Comment by zsliu98** [comment]
> @zsliu98 just downloaded. Yes, this process made the installation much more user friendly! Will definitely help others be able to successfully download the plug-in. I tried out the mono track issue and it is still the same unfortunately :/. I will continue using it with the playback playing since it shows the frequency band then. I tried out a stereo track to compare and also make sure everything was working for stereo track and stereo track worked fine.

Glad to hear that! However, I am afraid that I will not be able to solve the mono-track issue (cause all the curves are being updated while the audio is being processed). I am going to close this issue.

**Comment by rygantic217** [comment]
Hi just wanted to update you. I updated to Logic Pro Version 11 and the mono track is now working after you get playback audio to run through the plug in first. It still does not show the band when audio is not playing and you can only move the dot. However, if you play a section where there is audio, move the band and then pause the playback you can then move the frequency band when paused. Basically, it seems to be that the plug-in needs to “activate” by having the audio signal play, and then you can physically see the band and move it even when paused. But needs to start by having a section actually play audio first.

**Comment by rygantic217** [comment]
@zsliu98 ^^^ forgot to mention you

**Comment by zsliu98** [comment]
> @zsliu98 ^^^ forgot to mention you

@rygantic217 Thanks for the update. I'm glad that the plugin still works in the new Logic Pro! Yes, I think Logic Pro may disable the plugin processor when there is no audio playback (to save computation power?). Some commercial plugins can update the band curves even for high-order filters. However, I don't know how to do it efficiently ...
